src/Utilities/StdArrayHelpers.hpp

```diff
@@ -188,3 +188,59 @@ auto map_array(const std::array<T, Dim>& array, const F& f) noexcept {
   return std_array_helpers_detail::map_array_impl(
       array, f, std::make_index_sequence<Dim>{});
 }
+
+
+/**\ingroup UtilitiesGroup
+ * \brief Declares a binary function on an array, intended for operations
+ *
+ * \param RESULT_TYPE the array type that is the result of the operation (e.g A
+ * for resulting std::array<A,2>)
+ *
+ * \param LTYPE the array type of the first argument of the function (so the
+ * left value if the function is an operator overload)
+ *
+ * \param RTYPE the array type of the second argument of the function
+ *
+ * \param OP_FUNCTION_NAME the function which should be declared (e.g. operator+
+ * for an overload)
```

Comment:
I would just have left it at `operator+` since all operators will typically be overloaded :)